rest-service/manager_rest/test/endpoints/test_filters.py

```diff
@@ -2,181 +2,242 @@
-
-    def test_create_filters_mapping_fails(self):
-        with self.assertRaisesRegex(BadParametersError,
-                                    '.*have the same key.*'):
-            create_labels_filters_mapping(['a=b', 'b!=c', 'a=c'])
-
-    def test_create_filters_list(self):
-        created_filters_list = get_filters_list_from_mapping(
-            create_labels_filters_mapping(LEGAL_RULES))
-        self.assertEqual(set(created_filters_list), set(LEGAL_RULES))
-
-    def test_filters_applied(self):
+
+    def test_labels_filters_applied(self):
         dep1 = self.put_deployment_with_labels(self.LABELS)
         dep2 = self.put_deployment_with_labels(self.LABELS_2)
-        self._assert_filters_applied({EQUAL: {'a': ['b']}}, {dep1.id, dep2.id})
-        self._assert_filters_applied({NOT_EQUAL: {'c': ['z']}}, {dep1.id})
-        self._assert_filters_applied({EQUAL: {'a': ['y', 'z'], 'c': ['d']}},
+        self._assert_filters_applied([('a', ['b'], 'any_of', 'label')],
+                                     {dep1.id, dep2.id})
+        self._assert_filters_applied([('c', ['z'], 'not_any_of', 'label')],
+                                     {dep1.id})
+        self._assert_filters_applied([('a', ['y', 'z'], 'any_of', 'label'),
+                                      ('c', ['d'], 'any_of', 'label')],
                                      {dep1.id})
-        self._assert_filters_applied({EQUAL: {'a': ['b']}, IS_NOT_NULL: ['e']},
+        self._assert_filters_applied([('a', ['b'], 'any_of', 'label'),
+                                      ('e', [], 'is_not_null', 'label')],
                                      {dep2.id})
-        self._assert_filters_applied({EQUAL: {'a': ['b']}, IS_NULL: ['e']},
+        self._assert_filters_applied([('a', ['b'], 'any_of', 'label'),
+                                      ('e', [], 'is_null', 'label')],
                                      {dep1.id})
-        self._assert_filters_applied({IS_NULL: ['a']}, set())
+        self._assert_filters_applied([('a', [], 'is_null', 'label')], set())
         self._assert_filters_applied(
-            {EQUAL: {'a': ['b']}, NOT_EQUAL: {'c': ['y', 'z']}}, {dep1.id})
-
-    def test_filters_functionality_fails(self):
-        err_filters = ['a null', 'a', 'a!b']
-        for err_filter in err_filters:
-            with self.assertRaisesRegex(BadLabelsFilter,
-                                        '.*not in the right format.*'):
-                create_labels_filters_mapping([err_filter])
+            [('a', ['b'], 'any_of', 'label'),
+             ('c', ['y', 'z'], 'not_any_of', 'label')], {dep1.id})
+
+    def test_filter_rule_not_dictionary_fails(self):
+        with self.assertRaisesRegex(BadFilterRule, '.*not a dictionary'):
+            parse_filter_rules(['a'], models.Deployment)
+
+    def test_filter_rule_missing_entry_fails(self):
+        with self.assertRaisesRegex(BadFilterRule,
+                                    '.*one of the entries in the filter rule'
```

Comment:
```suggestion
                                    'one of the entries in the filter rule'
```

Reply:
and tbh i'd just make it `assertRaisesRegex(BadFilterRule, 'missing')`: it's a good idea to check that the message points to the part that is wrong, but no need to check the exact wording

Reply:
Thanks, changed.